### PR TITLE
add search bar and cleanup

### PR DESCRIPTION
**Pull request by SumitKPandit**

### CODE REVIEW

File: src/components/Dropdown/Dropdown.js [hunk 1]
```diff
@@ -27,11 +27,11 @@ const Dropdown = ({ buttonText, list, onSelect, selectedItem, theme }) => {
             `}
         >
             <div className={styles.clickArea} onClick={toggleDropdown}>
-                <span>{buttonText}</span>
+                <span>Filter by Region</span>
```
Comment (predeyo) [inline src/components/Dropdown/Dropdown.js:30]: Why did you do this? If you hard code the value component cannot be reused in other places in the app, if for example we would need an other dropdown to select language of a country we would need to make an other component for that as this one is hard coded for regions. This is bad idea.

File: src/utils.js [hunk 2]
```diff
@@ -15,9 +15,13 @@ const getCountries = () =>{
...
-    const countryNames = countries.map(country => country.name);
-    const found = countryNames.find(name => name === search);
-    return (found ? `found ${search}!` : `No results for ${search} found :(`);
+    const newCountries = countries.filter(country => {
+        return country.name.includes(search);
+    });
+    return newCountries;
+    // const countryNames = countries.map(country => country.name);
+    // const found = countryNames.find(name => name === search);
+    // return (found ? `found ${search}!` : `No results for ${search} found :(`);
```
Comment (predeyo) [inline src/utils.js:24]: Needs cleanup

File: src/components/CountryGrid/CountryGrid.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                            {country.nativeName} <br />/ population:{' '}
+                            {country.population} <br />/ region:{' '}
+                            {country.region} <br />/ subregion:{' '}
+                            {country.subregion} <br />/ capital:{' '}
+                            {country.capital} <br />/ top level domain:{' '}
+                            {country.topLevelDomain} <br />/ borders:{' '}
+                            {getBorders(country.borders)} <br />/ currencies:{' '}
+                            {getNestedData(country.currencies)} <br />/
+                            languages: {getNestedData(country.languages)}
+                        </li> */}
```
Comment (predeyo) [inline src/components/CountryGrid/CountryGrid.js:29]: Not needed.

File: src/App.js [hunk 4]
```diff
@@ -14,6 +12,16 @@ const App = () => {
     const [theme, setTheme] = useState('light');
     const [countries, setCountries] = useState([]);
     const [region, setRegion] = useState('');
+    const [searchTxt, setSearchTxt] = useState('');
+
+    // Search Form Handler
+    const searchHandler = e => {
+        setSearchTxt(e.target.value);
+        setCountries(searchByName(countries, searchTxt));
+    };
```
Comment (predeyo) [inline src/App.js:21]: Search is not working as we are constantly reducing the 'countries' state variable as we type in the search bar. Try to delete something from the search bar - no result, as you are passing in filtered list of countries in searchByName function.

File: src/components/Dropdown/Dropdown.js (hunk 1 above)
Comment (SumitKPandit) [reply src/components/Dropdown/Dropdown.js:30]: I was not aware if we have plans to create more filter options

**Comment by SumitKPandit** [comment]
Updated the PR with bug fix and review rework

File: src/utils.js [hunk 5]
```diff
@@ -13,12 +13,17 @@ const getCountries = () =>{
...
+// filter countries only for search text by name
 const searchByName = (countries, search) => {
-    const countryNames = countries.map(country => country.name);
-    const found = countryNames.find(name => name === search);
-    return (found ? `found ${search}!` : `No results for ${search} found :(`);
-
+    let newCountries = null;
+    if(search === "") {
+        newCountries = [...countries];
+        return newCountries;
```
Comment (predeyo) [inline src/utils.js:21]: Why so many steps? Why spread & assign & return? Just do `return countries` it is clean&clear.

File: src/components/Dropdown/Dropdown.js [hunk 6]
```diff
@@ -27,11 +27,11 @@ const Dropdown = ({ buttonText, list, onSelect, selectedItem, theme }) => {
             `}
         >
             <div className={styles.clickArea} onClick={toggleDropdown}>
-                <span>{buttonText}</span>
+                <span>{buttonTxt}</span>
```
Comment (predeyo) [inline src/components/Dropdown/Dropdown.js:30]: Why are you trying to shorten word "Text" to "Txt" it does not bring any value! I do understand what you mean by Txt but there is no reason for shortening a word by one letter and  potentially making it harder for others to read & understand the code.

File: src/App.js [hunk 7]
```diff
@@ -14,70 +12,73 @@ const App = () => {
     const [theme, setTheme] = useState('light');
     const [countries, setCountries] = useState([]);
     const [region, setRegion] = useState('');
+    const [searchTxt, setSearchTxt] = useState('');
```
Comment (predeyo) [inline src/App.js:15]: See comment for Txt is Dropdown.js

File: src/components/SearchBar/SearchBar.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+                theme={props.theme}
+                searchTxt={props.searchTxt}
+                searchHandler={props.searchHandler}
+            />
+            <Dropdown 
+                list={props.regionList}
+                theme={props.theme}
+                selectedItem={props.region}
+                onSelect={props.setRegion}
+                buttonTxt={props.buttonTxt}
```
Comment (predeyo) [inline src/components/SearchBar/SearchBar.js:20]: @ App.js there is no need to define a constant for a single label and then pass it down two components! Just do: `buttonText="Filter by Region"` right here and remove const in the App.

**Pull request merged**